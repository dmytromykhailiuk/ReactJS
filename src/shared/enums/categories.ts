export enum Categories {
  ALL = "ALL",
}
