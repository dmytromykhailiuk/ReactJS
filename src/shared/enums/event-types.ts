export enum EventTypes {
  CLICK = "click",
  SCROLL = "scroll",
}
