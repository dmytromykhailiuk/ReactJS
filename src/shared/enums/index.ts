export * from "./button-types";
export * from "./button-size";
export * from "./categories";
export * from "./sorting-options";
export * from "./movie-form-values";
export * from "./router-paths";
export * from "./modal-types";
