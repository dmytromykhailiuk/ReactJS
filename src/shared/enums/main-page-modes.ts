export enum MainPageModes {
  OVERVIEW = "overview",
  EDIT = "edit",
  CREATE = "create",
  DELETE = "delete",
}
