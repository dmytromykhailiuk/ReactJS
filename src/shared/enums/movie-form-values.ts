export enum MovieFormValues {
  MOVIE_ID = "id",
  TITLE = "title",
  RELEASE_DATE = "releaseDate",
  MOVIE_URL = "url",
  CATEGORY = "category",
  OVERVIEW = "overview",
  RUNTIME = "runtime",
  DURATION = "duration",
  RATING = "rating",
}
