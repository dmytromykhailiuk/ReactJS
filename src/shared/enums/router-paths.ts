export enum RouterPaths {
  HOME = '/',
  FILM = '/film/:id',
  SEARCH = '/search',
  ERROR = '/404',
}
