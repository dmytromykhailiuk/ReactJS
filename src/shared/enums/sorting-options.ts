export enum SortingOptionsProperties {
  RELEASE_DATE = "releaseDate",
  TITLE = "title",
}

export enum SortingOptions {
  releaseDate = "RELEASE DATE",
  title = "TITLE",
}
