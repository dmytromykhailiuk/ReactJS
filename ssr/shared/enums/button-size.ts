export enum ButtonSize {
  SMALL = "small",
  BIG = "big",
}
