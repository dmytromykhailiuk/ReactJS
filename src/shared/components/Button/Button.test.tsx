import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import { Button } from '.';
import { ButtonTypes, ButtonSize } from '../../enums';

describe('Button', () => {
  let wrapper: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;
  const onButtonClicked = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();

    wrapper = shallow(<Button>Button</Button>);
  });

  it('should match snepshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should add class 'button--secondary' when type equal 'ButtonTypes.SECONDARY'", () => {
    wrapper.setProps({ type: ButtonTypes.SECONDARY });
    expect(wrapper.exists('.button--secondary')).toBeTruthy();
  });

  it("should add class 'button--small' when size equal 'ButtonSize.SMALL'", () => {
    wrapper.setProps({ size: ButtonSize.SMALL, isSubmit: true });
    expect(wrapper.exists('.button--small')).toBeTruthy();
  });

  it('should call onClick when click on button', () => {
    wrapper.setProps({ onButtonClicked });
    wrapper.find('button').simulate('click');

    expect(onButtonClicked).toHaveBeenCalled();
  });
});
