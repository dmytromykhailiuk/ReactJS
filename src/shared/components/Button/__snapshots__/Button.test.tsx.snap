// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snepshot 1`] = `
<button
  className="
        button 
         
        button--primary 
        button--big 
      "
  disabled={false}
  onClick={[Function]}
  style={Object {}}
  type="button"
>
  Button
</button>
`;
